src/stream/src/executor/hash_agg.rs

```diff
@@ -353,12 +353,14 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
             call_visibilities.push(agg_call_filter_res);
         }
 
-        // Materialize input chunk if needed.
-        this.storages
-            .iter_mut()
+        // Materialize input chunk if needed and possible.
+        let materialized: Bitmap = this
+            .agg_calls
+            .iter()
+            .zip_eq_fast(&mut this.storages)
             .zip_eq_fast(call_visibilities.iter().map(Option::as_ref))
-            .for_each(|(storage, visibility)| {
-                if let AggStateStorage::MaterializedInput { table, mapping } = storage {
+            .map(|((call, storage), visibility)| {
+                if let AggStateStorage::MaterializedInput { table, mapping } = storage && !call.distinct {
```

Comment:
This is to fallback to maintaining MaterializedInputState per group if any call is distinct agg.